src/ddsrt/src/filesystem/posix/filesystem.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+#include <string.h>
+
+#include "dds/ddsrt/filesystem.h"
+#include "dds/ddsrt/string.h"
+#include "dds/ddsrt/heap.h"
+
+dds_return_t ddsrt_opendir(const char *name, ddsrt_dirHandle *dir)
+{
+    dds_return_t result;
+    DIR *d;
+
+    result = DDS_RETCODE_ERROR;
```

Comment:
It shouldn't come as a big surprise that I comment on the structure here: this could be almost a one-liner ... I wonder why not something like
```C
dds_return_t ddsrt_opendir(const char *name, ddsrt_dirHandle *dir)
{
  *dir = opendir (name);
  return *dir ? DDS_RETCODE_OK : DDS_RETCODE_OK;
}
```
(although I admit that it is ever so slightly different in that it overwrites ``dir`` on failure). Anyway, I would appreciate it if you went over this cleaning it up a bit.

And as ``ddsrt_dirHandle`` is typedef to a ``DIR *``, the type casts between the two are superfluous (and I would argue, therefore dangerous).